File: cu-manager/src/main/webapp/app/scripts/controllers/snapshot.js
/*
 * LICENCE : CloudUnit is available under the Affero Gnu Public License GPL V3 : https://www.gnu.org/licenses/agpl-3.0.html
 *     but CloudUnit is licensed too under a standard commercial license.
 *     Please contact our sales team if you would like to discuss the specifics of our Enterprise license.
 *     If you are not sure whether the GPL is right for you,
 *     you can always test our software under the GPL and inspect the source code before you contact us
 *     about purchasing a commercial license.
 *
 *     LEGAL TERMS : "CloudUnit" is a registered trademark of Treeptik and can't be used to endorse
 *     or promote products derived from this project without prior written permission from Treeptik.
 *     Products or services derived from this software may not be called "CloudUnit"
 *     nor may "Treeptik" or similar confusing terms appear in their names without prior written permission.
 *     For any questions, contact us : contact@treeptik.fr
 */

(function () {
  'use strict';

  /**
   * @ngdoc function
   * @name webuiApp.controller:SnapshotCtrl
   * @description
   * # SnapshotCtrl
   * Controller of the webuiApp
   */
  angular
    .module('webuiApp')
    .controller('SnapshotCtrl', SnapshotCtrl);

  SnapshotCtrl.$inject = [
    'SnapshotService',
    'ApplicationService',
    '$filter'
  ];

  function SnapshotCtrl(SnapshotService, ApplicationService, $filter) {
    var vm = this;
    vm.snapshots = [];
    vm.deleteTag = deleteTag;
    vm.cloneTag = cloneTag;
    vm.isValid = isValid;
    vm.applicationName = '';
    vm.message = '';
    vm.resetForm = resetForm;
    vm.errorMsg = false;

    init();

    ////////////////////////////////////////////

    function init() {
      SnapshotService.list()
        .then(success);

      function success(snapshots) {
        var decoratedSnapshots = [];
        angular.forEach(snapshots, function (snap) {
          snap.calendarDate = $filter('date')(snap.date, 'mediumDate');
          decoratedSnapshots.push(snap);
        });
        vm.snapshots = decoratedSnapshots;
      }
    }


    function cloneTag(snapshot) {
      SnapshotService.cloneTag(vm.applicationName, snapshot.tag)
        .then(success)
        .catch(error);

      function success() {
        init();
        resetForm();
      }

      function error(response){
        vm.errorMsg = response.data.message;
      }
    }

    function deleteTag(tag) {
      SnapshotService.deleteTag(tag)
        .then(success)
        .catch(error);

      function success() {
        init();
      }

      function error(response){
        vm.errorMsg = response.data.message;
      }
    }

    function isValid(applicationName, serverName) {
      ApplicationService.isValid(applicationName, serverName)
        .then(success)
        .catch(error);

      function success() {
        vm.message = '';
      }

      function error(response) {
        vm.message = response.data.message;
      }
    }

    function resetForm(){
      vm.cloneAppForm.$setPristine();
      vm.applicationName = '';
      vm.message = '';
    }
  }
})();

File: cu-cli/src/main/java/fr/treeptik/cloudunit/cli/model/Snapshot.java
/*
 * LICENCE : CloudUnit is available under the GNU Affero General Public License : https://gnu.org/licenses/agpl.html
 *     but CloudUnit is licensed too under a standard commercial license.
 *     Please contact our sales team if you would like to discuss the specifics of our Enterprise license.
 *     If you are not sure whether the GPL is right for you,
 *     you can always test our software under the GPL and inspect the source code before you contact us
 *     about purchasing a commercial license.
 *
 *     LEGAL TERMS : "CloudUnit" is a registered trademark of Treeptik and can't be used to endorse
 *     or promote products derived from this project without prior written permission from Treeptik.
 *     Products or services derived from this software may not be called "CloudUnit"
 *     nor may "Treeptik" or similar confusing terms appear in their names without prior written permission.
 *     For any questions, contact us : contact@treeptik.fr
 */

package fr.treeptik.cloudunit.cli.model;

import java.io.Serializable;
import java.util.Date;
import java.util.List;
import java.util.Map;

public class Snapshot implements Serializable {

    private static final long serialVersionUID = 1L;

    private Integer id;

    private String tag;
    
    private String displayTag;

    private String cuInstanceName;

    private String description;

    private String applicationName;
    
    private String applicationDisplayName;

    private String type;

    private String jvmRelease;

    private String jvmOptions;

    private Long jvmMemory;

    private User user;

    private Date date;

    private Map<String, String> appConfig;

    private List<String> images;

    private String deploymentStatus;

    private List<String> savedPorts;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTag() {
        return tag;
    }

    public void setTag(String tag) {
        this.tag = tag;
    }
    
    public String getDisplayTag() {
        return displayTag;
    }

    public void setDisplayTag(String displayTag) {
        this.displayTag = displayTag;
    }

    public String getCuInstanceName() {
        return cuInstanceName;
    }

    public void setCuInstanceName(String cuInstanceName) {
        this.cuInstanceName = cuInstanceName;
    }

    public String getApplicationName() {
        return applicationName;
    }

    public void setApplicationName(String applicationName) {
        this.applicationName = applicationName;
    }
    
    public String getApplicationDisplayName() {
        return applicationDisplayName;
    }

    public void setApplicationDisplayName(String applicationDisplayName) {
        this.applicationDisplayName = applicationDisplayName;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Map<String, String> getAppConfig() {
        return appConfig;
    }

    public void setAppConfig(Map<String, String> appConfig) {
        this.appConfig = appConfig;
    }

    public List<String> getImages() {
        return images;
    }

    public void setImages(List<String> images) {
        this.images = images;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getJvmRelease() {
        return jvmRelease;
    }

    public void setJvmRelease(String jvmRelease) {
        this.jvmRelease = jvmRelease;
    }

    public String getJvmOptions() {
        return jvmOptions;
    }

    public void setJvmOptions(String jvmOptions) {
        this.jvmOptions = jvmOptions;
    }

    public Long getJvmMemory() {
        return jvmMemory;
    }

    public void setJvmMemory(Long jvmMemory) {
        this.jvmMemory = jvmMemory;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDeploymentStatus() {
        return deploymentStatus;
    }

    public void setDeploymentStatus(String deploymentStatus) {
        this.deploymentStatus = deploymentStatus;
    }

    public List<String> getSavedPorts() {
        return savedPorts;
    }
}
